src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -263,30 +279,55 @@ def parse_quantization_config(
 
         return quantization_config
 
+    def _fetch_unique_quantization_formats(self) -> List[str]:
+        """
+        Get all unique compression formats present in a model
+        :return: list of quantization formats
+        """
+        quantization_formats = []
+        for _, scheme in self.quantization_config.config_groups.items():
+            if scheme.format is not None and scheme.format not in quantization_formats:
+                quantization_formats.append(scheme.format)
+
+        # If empty list, fallback to using the global format
+        if len(quantization_formats) == 0:
+            quantization_formats.append(self.quantization_config.format)
+        return quantization_formats
+
     def __init__(
         self,
         sparsity_config: Optional[SparsityCompressionConfig] = None,
         quantization_config: Optional[QuantizationConfig] = None,
         transform_config: Optional[TransformConfig] = None,
+        compression_formats: Optional[List[str]] = None,
     ):
         self.sparsity_config = sparsity_config
         self.quantization_config = quantization_config
         self.transform_config = transform_config
+        self.compression_formats = compression_formats
 
         self.sparsity_compressor = None
         self.quantization_compressor: Optional[
-            Union[BaseQuantizationCompressor, DenseCompressor]
+            Dict[str, Union[BaseQuantizationCompressor, DenseCompressor]]
         ] = None
         # no transform compressor is required
 
         if sparsity_config is not None:
             self.sparsity_compressor = BaseCompressor.load_from_registry(
                 sparsity_config.format, config=sparsity_config
             )
+
         if quantization_config is not None:
-            self.quantization_compressor = BaseCompressor.load_from_registry(
-                quantization_config.format, config=quantization_config
-            )
+            if not self.compression_formats:
+                self.compression_formats = self._fetch_unique_quantization_formats()
+
+            self.quantization_compressor = {}
+            for format in self.compression_formats:
+                self.quantization_compressor[
+                    format
+                ] = BaseCompressor.load_from_registry(
+                    format, config=quantization_config
```

Comment:
Now that a compressor is local to a `config_group`, but `quantization_config` is global, this no longer seems valid. Fortunately, the config is only ever referenced in `Sparse24BitMaskCompressor` [here](https://github.com/neuralmagic/compressed-tensors/blob/main/src/compressed_tensors/compressors/sparse_compressors/sparse_24_bitmask.py#L57), so maybe this is ok for now

Reply:
I’m confused - what is no longer valid?

Reply:
quantization_config is the global config, but we're passing it into a compressor that is scoped to a subset of modules. afaict there's no way for the compressor to backtrack which part of the `quantization_config` it is tied to. but `quantization_config` doesn't ever seem to be used anywhere, `self.config` is only ever referenced in `Sparse24BitMaskCompressor`. so it is only a problem if we use `self.config` in future commits. I don't think we can just remove it entirely because of its use in `Sparse24BitMaskCompressor`

Reply:
What you're referring to I believe is the SparsityConfig, not the QuantizationConfig: https://github.com/neuralmagic/compressed-tensors/blob/33c52def62ac44647fe924a05ed5115abfb0fdfd/src/compressed_tensors/config/base.py#L95

Generally speaking the quant config shouldnt be required in the compressor classes /  should be removed